packages/headless/src/app/commerce-engine/commerce-engine.ts

```diff
@@ -58,10 +59,21 @@ export type CommerceEngineState =
  * @internal WORK IN PROGRESS. DO NOT USE IN ACTUAL IMPLEMENTATIONS.
  */
 export interface CommerceEngine<State extends object = {}>
-  extends CoreEngine<
-    State & CommerceEngineState,
-    CommerceThunkExtraArguments
-  > {}
+  extends CoreEngine<State & CommerceEngineState, CommerceThunkExtraArguments> {
+  /**
+   * Executes the first search.
+   *
+   * @param analyticsEvent - The analytics event to log in association with the first search. If unspecified, `logInterfaceLoad` will be used.
+   */
+  executeFirstSearch(): void;
```

Comment:
Can we use the commerce search controller instead? Or do we absolutely need these methods to be exposed on the engine?

Reply:
I think for now, we're working in mimetism to what's in the `search-engine.ts.`

Worth considering in another issue/❓/PR

Reply:
I also think that this is a nice-to-have, as the Engine is intrinsically linked to the search-interface today.

Subject to changes tho